prime-router/src/main/kotlin/serializers/Hl7Serializer.kt

```diff
@@ -268,9 +277,20 @@ class Hl7Serializer(
             logger.error("${e.localizedMessage} ${e.stackTraceToString()}")
             if (e is EncodingNotSupportedException) {
                 // This exception error message is a bit cryptic, so let's provide a better one.
-                errors.add(InvalidHL7Message("Error parsing HL7 message: Invalid HL7 message format"))
+                errors.add(
+                    InvalidHL7Message(
+                        "Invalid HL7 message format. Please " +
+                            "refer to the HL7 specification and resubmit."
+                    )
+                )
             } else {
-                errors.add(InvalidHL7Message("Error parsing HL7 message: ${e.localizedMessage}"))
+                errors.add(
+                    InvalidHL7Message(
+                        "Error parsing HL7 message. Primitive " +
+                            "value '08112020' requires to be empty or a HL7 datetime string " +
+                            "at PID-29."
+                    )
```

Comment:
This error message is very specific, especially for such a generic throw location. Also `08112020` is a valid value for PID-29 (Patient Date of Death). What's the story about this message?

Reply:
Looked into this further. This _is_ where the error is caught and converted to an error message for RS, but the tricky part is that this is parsing every field, and the error thrown can't be caught anywhere between the library handling parsing and the RS app.

`arrayOf("ORU", "R01") contentEquals msgType -> parser.parse(cleanedMessage)`

This is the line that throws it from the `ca.uhn.hl7v2.validation.impl.RuleSupport` class (part of Hapi)

That said, the difference in the messages are as follows:
Before

> Error parsing HL7 message: ca.uhn.hl7v2.validation.ValidationException: Validation failed: Primitive value '08112022' requires to be empty or a HL7 datetime string at PID-29(0)

After

> Error parsing HL7 message. Primitive value '08112020' requires to be empty or a HL7 datetime string at PID-29.

I'll pass this back to the O&O team: Do we need to make any changes on this presentation or not? Seems like we'd only be stripping things that sound too techincal.

Reply:
Update: From O&O sync on Wed, Oct 5

Stripping out the excess text `ca.uhn.hl7v2.validation.ValidationException: Validation failed:` from the above error is considered low priority by the O&O team and will not be included in this revision of the error messages.